src/cascadia/TerminalSettingsEditor/Appearances.h

```diff
@@ -45,13 +46,88 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 
         hstring ToString() { return _LocalizedName; }
         bool HasPowerlineCharacters();
+        Windows::Foundation::Collections::IMap<winrt::hstring, winrt::hstring> FontAxesTagsAndNames();
 
         WINRT_PROPERTY(hstring, Name);
         WINRT_PROPERTY(hstring, LocalizedName);
 
     private:
         winrt::com_ptr<IDWriteFontFamily> _family;
         std::optional<bool> _hasPowerlineCharacters;
+        winrt::hstring _axisTagToString(DWRITE_FONT_AXIS_TAG tag);
+    };
+
+    struct AxisKeyValuePair : AxisKeyValuePairT<AxisKeyValuePair>
+    {
+        AxisKeyValuePair(winrt::hstring axisKey, float axisValue, const Windows::Foundation::Collections::IMap<winrt::hstring, float>& baseMap, const Windows::Foundation::Collections::IMap<winrt::hstring, winrt::hstring>& tagToNameMap) :
+            _AxisKey{ axisKey },
+            _AxisValue{ axisValue },
+            _baseMap{ baseMap },
+            _tagToNameMap{ tagToNameMap }
+        {
+            if (_tagToNameMap.HasKey(_AxisKey))
+            {
+                int32_t i{ 0 };
+                // this loop assumes that every time we iterate through the map
+                // we get the same ordering - is this true?
+                for (const auto tagAndName : _tagToNameMap)
+                {
+                    if (tagAndName.Key() == _AxisKey)
+                    {
+                        _AxisIndex = i;
+                        break;
+                    }
+                    ++i;
+                }
```

Comment:
This is the manual 'index to key' mapping we have to do because xaml struggles with combo box two-way bindings